### PR TITLE
[CARBONDATA-2098] Add datamap managment description

### PR DESCRIPTION
**Pull request by jackylk**
Enhance document for datamap

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2809/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4055/

File: docs/datamap/preaggregate-datamap-guide.md [hunk 1]
```diff
@@ -193,8 +230,10 @@ main table but not performed on pre-aggregate table, all queries still can benef
 pre-aggregate tables. To further improve the query performance, compaction on pre-aggregate tables 
 can be triggered to merge the segments and files in the pre-aggregate tables. 
 
-#### Data Management on pre-aggregate tables
-Once there is pre-aggregate table created on the main table, following command on the main table
+## Data Management with pre-aggregate tables
+In current implementation, data consistence need to maintained for both main table and pre-aggregate
```
Comment (ravipesala) [inline docs/datamap/preaggregate-datamap-guide.md]: typo `need to be maintained`

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4058/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2812/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**